### PR TITLE
added genres and search by genre function

### PR DESCRIPTION
**Pull request by arbone**
I added all the genres to all books and to let people find books by genres I added a specific function as my other colleagues did. Although I have some problem with the function because I tried but it does not run perfectly. All the data is correct but still I have problems.

### CODE REVIEW

**Comment by arbone** [comment]
Maybe I found out the error, I forgot to add the line "from library import check_genre" (which I added later on)... this can be the answer but I am not quite sure. So I think I need help on this on.

**Review by FPolin22 — COMMENTED** [review]
I tried to correct it on my own, I find a mistake and you can correct it by just deleting it from line 131 to 133 (last if function), if you can't run it correctly I will send you the correct pull request

File: library.py [hunk 1]
```diff
@@ -106,6 +108,26 @@ def check_year(year):
...
         pub_year = field[1]
+        genre = field[2]
+        if year == pub_year:
+            print("{} published {} in {} ".format(author, title, year))
+            found = True
+
+    if not found:
+        print("Sorry we don't have any books published in the year {} ".format(year))
+        
+def check_genre(book_genre):
```
Comment (Breach97) [inline library.py:119]: Are you sure that `book_genre` in the parenthesis is correct?  Try `genre` and see if it fix the bug

File: library.py [hunk 2]
```diff
@@ -106,6 +108,26 @@ def check_year(year):
...
+        author = field[0]
+        pub_year = field[1]
+        genre = field[2]
+        if genre == book_genre:
+            print("{} of {} is a {} ".format(title, author, genre))
+            found = True
+
+    if not found:
+        print("Sorry there aren't books in the {} category, try again".format(genre))
         if year == pub_year:
```
Comment (Breach97) [inline library.py:131]: **From line 129 to 131**  lines of code are the repetition of lines 112-114. you should **delete** them

**Review by Breach97 — CHANGES_REQUESTED** [review]
Made 2 comments, check them out at lines 119 and 129

File: library.py (hunk 2 above)
Comment (arbone) [reply library.py:131]: Thanks, this should be it

File: library.py (hunk 1 above)
Comment (arbone) [reply library.py:119]: Yes I think it is correct as I followed the other functions

Comment (arbone) [reply library.py:119]: But I accept any suggestions

Comment (Breach97) [reply library.py:119]: I think that is the error, you can test it locally to see if it works (y)

**Comment by FPolin22** [comment]
In local, it works perfectly

**Comment by arbone** [comment]
I had some issues with the code because in my files I had some additional code inserted by mistake from another team member, it was absolutely not a problem. I solved it by myself. Now I will try it locally and the write a comment here.

File: library.py (hunk 1 above)
Comment (arbone) [reply library.py]: There wasn't any error there, but thanks for the suggestion

**Comment by arbone** [comment]
Tried the code locally and it works. Ready to merge

**Comment by arbone** [comment]
Waiting for the “ok to merge” from both of my groupmates to merge it. When I receive it, I will merge it.

**Pull request merged**